src/userspace/exec_monitor.hpp

```diff
@@ -0,0 +1,20 @@
+#ifndef EXEC_MONITOR_H
+#define EXEC_MONITOR_H
+
+#include <vector>
+#include <string>
+
+class ExecMonitor
```

Comment:
In the future we will need more monitors, and they will most likely all do the same thing. It would be helpful to define a `Monitor` abstract class that all monitors would inherit so that it enforces that they can all be used in the same way.

Reply:
Done. However, I identified 4 common methods for the monitor classes, i.e. run(), export_output(), parse_args() and process_sample(). But I couldn't add two of them in the abstract class:
- parse_args: it is actually defined in config.cpp for each usecase, because this is where it is used
- process_sample: it should be a static function, cannot take the virtual keyword

So the abstract class has just run() and export_output().